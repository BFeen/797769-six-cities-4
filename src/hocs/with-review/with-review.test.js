import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import withReview from "./with-review.js";
import NameSpace from "../../reducer/name-space.js";


const mockStore = configureStore([]);

const MockComponent = () => {
  return (
    <div></div>
  );
};

MockComponent.propTypes = {};

const MockComponentWrapped = withReview(MockComponent);

const userDataMock = {
  id: 0,
  name: `Max`,
  avatar: `img/img.png`,
  isPro: false,
  email: `max@mail.ru`,
};

describe(`withReview HOC snapshot testing`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      reviews: [],
    },
    [NameSpace.APPLICATION]: {
      offerId: 3,
    },
    [NameSpace.USER]: {
      authorizationStatus: `AUTH`,
      user: userDataMock,
    }
  });

  it(`withReview HOC rendering correctly`, () => {
    const tree = renderer.create(
        <Provider store={store}>
          <MockComponentWrapped
            offerId={0}
            handleSubmit={() => {}}
          />
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
