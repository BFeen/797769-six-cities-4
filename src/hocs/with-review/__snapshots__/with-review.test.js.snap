// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withReview HOC snapshot testing withReview HOC rendering correctly 1`] = `<div />`;
